repos/system_upgrade/common/actors/scansourcefiles/tests/unit_test_scansourcefiles.py

```diff
@@ -1,5 +1,163 @@
+        if raise_error:
+            raise CalledProcessError("mocked error", cmd, "result")
+        return run_output
+
+    raise_error = run_output and not isinstance(run_output, dict)
+    input_file = os.path.join(FILES_DIR, 'mocked_file')
+
+    monkeypatch.setattr(scansourcefiles, 'run', mocked_run)
+    monkeypatch.setattr(api, 'current_logger', testutils.logger_mocked())
+
+    assert scansourcefiles._get_rpm_name(input_file) == expected_output
+
+    if raise_error:
+        return
+
+    if len(run_output['stdout']) > 1:
+        assert len(api.current_logger.warnmsg) == 1
+        expected_warnmsg = 'The {} file is owned by multiple rpms: rpm1, rpm2.'.format(input_file)
+        assert api.current_logger.warnmsg[0] == expected_warnmsg
+    else:
+        assert not api.current_logger.warnmsg
+
+
+@pytest.mark.parametrize('input_file,rpm_name,modified,expected_output', (
+    (os.path.join(FILES_DIR, 'file_not_existant'), 'rpm', True,
+     FileInfo(path=os.path.join(FILES_DIR, 'file_not_existant'), exists=False, rpm_name='rpm', is_modified=True)),
+    (os.path.join(FILES_DIR, 'file_not_rpm_owned'), '', False,
+     FileInfo(path=os.path.join(FILES_DIR, 'file_not_rpm_owned'), exists=True, rpm_name='', is_modified=False))
+))
+def test_scan_file(monkeypatch, input_file, rpm_name, modified, expected_output):
+    monkeypatch.setattr(scansourcefiles, 'is_modified', lambda _: modified)
+    monkeypatch.setattr(scansourcefiles, '_get_rpm_name', lambda _: rpm_name)
+
+    file_info = scansourcefiles.scan_file(input_file)
+
+    assert isinstance(file_info, FileInfo)
+    assert file_info == expected_output
+
+
+@pytest.mark.parametrize('input_files,expected_output', (
+    ([], []),
+    (['file1'], [FileInfo(path='file1', exists=False, rpm_name='', is_modified=False)]),
+    (['file1', 'file2'], [FileInfo(path='file1', exists=False, rpm_name='', is_modified=False),
+                          FileInfo(path='file2', exists=False, rpm_name='', is_modified=False)]),
+))
+def test_scan_files(monkeypatch, input_files, expected_output):
+    def mocked_scan_file(input_file):
+        return FileInfo(path=input_file, exists=False, rpm_name='', is_modified=False)
+
+    monkeypatch.setattr(scansourcefiles, 'scan_file', mocked_scan_file)
+
+    assert scansourcefiles.scan_files(input_files) == expected_output
+
+
+@pytest.mark.parametrize('input_file,rpm_name,expected_output', (
+    (os.path.join(FILES_DIR, 'file_rpm_owned'), 'rpm',
+     FileInfo(path=os.path.join(FILES_DIR, 'file_rpm_owned'), exists=True, rpm_name='rpm', is_modified=False)),
+))
+def test_rpm_owned_files(monkeypatch, input_file, rpm_name, expected_output):
+    monkeypatch.setattr(api, 'produce', testutils.produce_mocked())
+    monkeypatch.setattr(scansourcefiles, 'get_source_major_version', lambda: 9)
+    monkeypatch.setattr(scansourcefiles, '_get_rpm_name', lambda _: rpm_name)
+    monkeypatch.setattr(scansourcefiles, 'is_modified', lambda _: False)
```

Comment:
Mock the scan_files function directly and return what it is expected to return. You can do that because that one is already covered by the test above.

Also, you shouldn't need to mock the TRACKED_FILES, since all calls operating directly on files can be mocked, e.g. the os.path.exists call (as explained in one of my other comments).